From f881080a13f5035a2d4c088a1ce999d6d38c5b0c Mon Sep 17 00:00:00 2001
From: Avi Niddam <aniddam@habana.ai>
Date: Thu, 05 Jun 2025 12:44:43 +0300
Subject: [PATCH] [SW-228856] fit: add fit_services to bld and use

We encountered several code duplications related to
fit_foreach_property.
So, we want to create a new fit services infrastructure
to solve this issue in a clean way.
The main Lego pieces are the newly created files:
fit_services.c/h
In this current commit we build in Preboot the newly
created fit_services.c and in module_lkd invoke
the function with its new temporary name.

General remark:
This infrastructure is intended to support ANY of the
fit related functions that we find today in fit_utils.c
or fit.c
Therefore, from now, please, anyone doing a change related
to the fit files should carefully double-check:
Any unified code can be (and should be) located in
fit_services.c/h
For this, please, follow the example of how it has been
done with fit_foreach_property.

Change-Id: I25a2027d85af64ca141e69a830617d56452043a1
Signed-off-by: Avi Niddam <aniddam@habana.ai>
---

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d8a841e..c9d4182 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -115,6 +115,7 @@
 zephyr_library_sources(emb_common_mod/lib/common/lib_common.c)
 zephyr_library_sources(emb_common_mod/lib/common/boot_desc.c)
 zephyr_library_sources(emb_common_mod/lib/common/pci.c)
+zephyr_library_sources(emb_common_mod/lib/common/fit_services.c)
 zephyr_library_sources(emb_common_mod/lib/preboot/fit.c)
 zephyr_library_sources(emb_common_mod/lib/preboot/fit_rsa.c)
 zephyr_library_sources(emb_common_mod/lib/preboot/fit_regions.c)
diff --git a/lib/module/module_lkd.c b/lib/module/module_lkd.c
index 5479d50..ce24552 100644
--- a/lib/module/module_lkd.c
+++ b/lib/module/module_lkd.c
@@ -27,6 +27,7 @@
 #include "fit.h"
 #include "security.h"
 #include <watchdog_timer.h>
+#include "fit_services.h"
 
 #define BIT_ULL(nr)	(1ULL << (nr))
 
@@ -308,7 +309,7 @@
 	int rc = 0;
 
 	fit_required_device_name = NULL;
-	fit_foreach_property((void *)(desc->base), "/", match_fit_prop_device, HL_FIT_DEVICE_PROP);
+	_fit_foreach_property((void *)(desc->base), "/", match_fit_prop_device, HL_FIT_DEVICE_PROP);
 	if (fit_required_device_name) {
 		if (memcmp(fit_required_device_name, HL_DEVICE_TYPE, strlen(HL_DEVICE_TYPE))) {
 			LOG_ERR("Attempt to load the FIT, that is intended for device %s",
